db/sql/0_msar.sql

```diff
@@ -2166,3 +2274,71 @@ BEGIN
   RETURN added_table_id;
 END;
 $$ LANGUAGE plpgsql RETURNS NULL ON NULL INPUT;
+
+
+----------------------------------------------------------------------------------------------------
+----------------------------------------------------------------------------------------------------
+-- TABLE SPLITTING FUNCTIONS
+--
+-- Functions to extract columns from a table
+----------------------------------------------------------------------------------------------------
+----------------------------------------------------------------------------------------------------
+
+
+CREATE OR REPLACE FUNCTION
+msar.extract_columns_from_table(
+  tab_id oid, col_ids integer[], new_tab_name text, fk_col_name text
+) RETURNS jsonb AS $f$/*
+Extract columns from a table to create a new table, linked by a foreign key.
+
+Args:
+  tab_id: The OID of the table whose columns we'll extract
+  col_ids: An array of the attnums of the columns to extract
+  new_tab_name: The name of the new table to be made from the extracted columns, unquoted
+  fk_col_name: The name to give the new foreign key column in the remainder table (optional)
+
+The extraction takes a set of columns from the table, and creates a new table from the set of
+*distinct* tuples those columns comprise. We also add a new foreign key column to the original
+ (remainder) table that links it to the new extracted table so they can be easily rejoined. The
+ extracted columns are removed from the remainder table.
+*/
+DECLARE
+  extracted_col_defs CONSTANT jsonb := msar.get_extracted_col_def_jsonb(tab_id, col_ids);
+  fkey_name CONSTANT text := msar.build_unique_fkey_column_name(tab_id, fk_col_name, new_tab_name);
+  extracted_table_id integer;
+  fkey_attnum integer;
+BEGIN
+  extracted_table_id := msar.add_mathesar_table(
+    msar.get_relation_namespace_oid(tab_id),
+    new_tab_name,
+    extracted_col_defs,
+    null,
+    format('Extracted from %s', __msar.get_relation_name(tab_id))
+  );
+  fkey_attnum := msar.create_many_to_one_link(extracted_table_id, tab_id, fkey_name);
+  PERFORM __msar.exec_ddl($t$
+    WITH fkey_cte AS (
+      SELECT id, %1$s, dense_rank() OVER (ORDER BY %1$s) AS __msar_tmp_id
```

Comment:
It will be helpful to add comments on what the placeholders refer to. Something like 
```
-- %1$s refers to the columns to be moved
```

Reply:
Done; I also added a number of other comments describing the process.